pkg/controller/instance/instance_controller.go

```diff
@@ -376,24 +423,46 @@ type ReconcileInstance struct {
 // +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=kudo.dev,resources=instances,verbs=get;list;watch;create;update;patch;delete
 func (r *ReconcileInstance) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx := context.TODO()
 	// Fetch the Instance instance
-	instance := &kudov1alpha1.Instance{}
-	err := r.Get(context.TODO(), request.NamespacedName, instance)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Object not found, return.  Created objects are automatically garbage collected.
-			// For additional cleanup logic use finalizers.
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
+	// Fetch the Instance instance
+	instance, err := getInstance(ctx, r, request)
+	if err != nil || instance == nil {
+		return reconcile.Result{}, err
+	}
+
+	// Make sure the OperatorVersion is present
+	ov, err := getOperatorVersion(ctx, r, r.recorder, instance)
+	if err != nil || ov == nil {
 		return reconcile.Result{}, err
 	}
 
 	log.Printf("InstanceController: Received Reconcile request for instance \"%+v\"", request.Name)
 
-	// Make sure the OperatorVersion is present
-	ov := &kudov1alpha1.OperatorVersion{}
-	err = r.Get(context.TODO(),
+	// if this is new create and create and assign a planexecution and return
+	if isNewInstance(instance) {
+		err = createPlan(r, "deploy", instance)
```

Comment:
I would actually like to have the check for deploy plan exist even here, I feel like now the fact that deploy is executed for these instances is spread in two places, while it can be in one...

Reply:
also considering what the method does, create plan should rather be `executePlan`. I believe that method should also check that plan exists, to be complete...

Reply:
`createPlan` literally creates the PlanExecution object for this instance in the `createPlanExecution` method that this method calls.   This doesn't execute the plan as much as it creates a planexecution that did not previous exist in the cluster.